api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -92,6 +98,48 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 	}
 
 	/**
+	 * @see org.openmrs.api.OrderService#saveOrders(Set)
+	 */
+	public synchronized Set<Order> saveOrders(Set<Order> orders) {
+		Set<OrderGroup> orderGroups = new LinkedHashSet<OrderGroup>();
+
+		for (Order o : orders) {
+			if (o.getOrderId() == null) {
+				if (o.getOrderGroup() != null) {
+					o.setOrderGroup(getOrderGroup(orderGroups, o.getOrderGroup()));
+					setSortWeightForOrders(o);
+				}
+				saveOrder(o, null);
+			}
+		}
+		return orders;
+	}
+
+
+
+	private OrderGroup getOrderGroup(Set<OrderGroup> orderGroups, OrderGroup newOrderGroup){
+		if(newOrderGroup.getId() != null){
+			orderGroups.add(Context.getOrderService().getOrderGroupById(newOrderGroup.getId()));
+		}
+		for (OrderGroup orderGroup : orderGroups) {
+			if (orderGroup.getOrderSet().equals(newOrderGroup.getOrderSet())) {
+				return orderGroup;
+			}
+		}
+		dao.saveOrderGroup(newOrderGroup);
+		orderGroups.add(newOrderGroup);
+		return newOrderGroup;
+	}
+
+
+	private void setSortWeightForOrders(Order o) {
+		for (OrderSetMember orderSetMember : o.getOrderGroup().getOrderSet().getOrderSetMembers()) {
+			if (o.getConcept().equals(orderSetMember.getConcept()))
```

Comment:
So the assumption here is that an order set can't have set members of the same concept, is there any validation to guarantee this when creating an order set?